cloud/services/publicloadbalancers/publicloadbalancers.go

```diff
@@ -72,89 +76,94 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
-				FrontendIPConfigurations: &[]network.FrontendIPConfiguration{
-					{
-						Name: &frontEndIPConfigName,
-						FrontendIPConfigurationPropertiesFormat: &network.FrontendIPConfigurationPropertiesFormat{
-							PrivateIPAllocationMethod: network.Dynamic,
-							PublicIPAddress:           &publicIP,
-						},
+	lb := network.LoadBalancer{
+		Sku:      &network.LoadBalancerSku{Name: network.LoadBalancerSkuNameStandard},
+		Location: to.StringPtr(s.Scope.Location()),
+		Tags: converters.TagsToMap(infrav1.Build(infrav1.BuildParams{
+			ClusterName: s.Scope.Name(),
+			Lifecycle:   infrav1.ResourceLifecycleOwned,
+			Role:        to.StringPtr(publicLBSpec.Role),
+			Additional:  s.Scope.AdditionalTags(),
+		})),
+		LoadBalancerPropertiesFormat: &network.LoadBalancerPropertiesFormat{
+			FrontendIPConfigurations: &[]network.FrontendIPConfiguration{
+				{
+					Name: &frontEndIPConfigName,
+					FrontendIPConfigurationPropertiesFormat: &network.FrontendIPConfigurationPropertiesFormat{
+						PrivateIPAllocationMethod: network.Dynamic,
+						PublicIPAddress:           &publicIP,
 					},
 				},
-				BackendAddressPools: &[]network.BackendAddressPool{
-					{
-						Name: &backEndAddressPoolName,
-					},
-				},
-				Probes: &[]network.Probe{
-					{
-						Name: &probeName,
-						ProbePropertiesFormat: &network.ProbePropertiesFormat{
-							Protocol:          network.ProbeProtocolTCP,
-							Port:              to.Int32Ptr(s.Scope.APIServerPort()),
-							IntervalInSeconds: to.Int32Ptr(15),
-							NumberOfProbes:    to.Int32Ptr(4),
-						},
-					},
+			},
+			BackendAddressPools: &[]network.BackendAddressPool{
+				{
+					Name: &backEndAddressPoolName,
 				},
-				// We disable outbound SNAT explicitly in the HTTPS LB rule and enable TCP and UDP outbound NAT with an outbound rule.
-				// For more information on Standard LB outbound connections see https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-outbound-connections.
-				LoadBalancingRules: &[]network.LoadBalancingRule{
-					{
-						Name: to.StringPtr("LBRuleHTTPS"),
-						LoadBalancingRulePropertiesFormat: &network.LoadBalancingRulePropertiesFormat{
-							DisableOutboundSnat:  to.BoolPtr(true),
-							Protocol:             network.TransportProtocolTCP,
-							FrontendPort:         to.Int32Ptr(s.Scope.APIServerPort()),
-							BackendPort:          to.Int32Ptr(s.Scope.APIServerPort()),
-							IdleTimeoutInMinutes: to.Int32Ptr(4),
-							EnableFloatingIP:     to.BoolPtr(false),
-							LoadDistribution:     network.LoadDistributionDefault,
-							FrontendIPConfiguration: &network.SubResource{
+			},
+			OutboundRules: &[]network.OutboundRule{
+				{
+					Name: to.StringPtr("OutboundNATAllProtocols"),
+					OutboundRulePropertiesFormat: &network.OutboundRulePropertiesFormat{
+						Protocol:             network.LoadBalancerOutboundRuleProtocolAll,
+						IdleTimeoutInMinutes: to.Int32Ptr(4),
```

Comment:
In aks engine this is set to 30 mins: 

https://github.com/Azure/aks-engine/blob/95d6127337008ed562d3723a53dbffbc99b2eee2/pkg/api/defaults.go#L421
https://github.com/Azure/aks-engine/blob/108dfabdaaf049b3062d9b6298e4ca03b9b3473e/pkg/api/const.go#L229

Maybe @jackfrancis can confirm?

Reply:
Correct, and it's also user-configurable. I can't speak to whether 4 or 30 is a preferable default. I do wonder if it should be user-configurable.

What does this outbound rule timeout govern?

Reply:
4 is the default value https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-outbound-connections#idletimeout and is also the same value CAPZ is already using so didn't want to change the default on this PR (this block of code just moved down).
We can look at making it configurable if there is a use case for configuring that value separately.

Reply:
The comment above the linked line seems like it might have been a bug at some point?

>// DefaultOutboundRuleIdleTimeoutInMinutes determines the aks-engine provided default for IdleTimeoutInMinutes of the OutboundRule of the agent loadbalancer
	// This value is set greater than the default Linux idle timeout (15.4 min): https://pracucci.com/linux-tcp-rto-min-max-and-tcp-retries2.html

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/676

Reply:
Understood. I'm not sure why the default is set so high in AKS Engine, as it would seem to have the side-effect of reducing port availability (leading to exhaustion).

In any event, having read about what this does I think the proper value of this is going to be very much app domain-specific, so: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/677